### PR TITLE
Fix hardcoded max data volumes when VM has been created but not started before

### PR DESCRIPTION
**Pull request by nvazquez**
## Description
When VM is created but not started on any host, attaching a volume on it causes the maximum number of allowed volumes to be 6 (hardcoded)

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
- Deploy VM with startvm=false parameter
- Attach more than 6 volumes to VM (in case of KVM and VMware)

### CODE REVIEW

**Comment by nvazquez** [comment]
Ping @anuragaw @shwstppr can you please review?

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 1]
```diff
@@ -3007,6 +3009,13 @@ private int getMaxDataVolumesSupported(UserVmVO vm) {
         if (host != null) {
             _hostDao.loadDetails(host);
             maxDataVolumesSupported = _hypervisorCapabilitiesDao.getMaxDataVolumesLimit(host.getHypervisorType(), host.getDetail("product_version"));
+        } else {
+            List<HypervisorType> supportingDefaultHV = Arrays.asList(HypervisorType.XenServer, HypervisorType.VMware,
+                    HypervisorType.KVM, HypervisorType.Ovm, HypervisorType.LXC);
+            HypervisorType hypervisorType = vm.getHypervisorType();
+            if (hypervisorType != null && supportingDefaultHV.contains(hypervisorType)) {
+                maxDataVolumesSupported = _hypervisorCapabilitiesDao.getMaxDataVolumesLimit(hypervisorType, "default");
```
Comment (rhtyd) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java]: Could version specific values cause regression or bug?

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java]: if these are lower the "default", yes @rhtyd. But that should never be the case, should it?

**Comment by DaanHoogland** [comment]
hudson couldn't reach github :(

**Comment by DaanHoogland** [comment]
try again

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java (hunk 1 above)
Comment (nvazquez) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java]: Good point, I've verified any version contains max limit lower than the default for each hypervisor

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

File: engine/schema/src/main/java/com/cloud/hypervisor/dao/HypervisorCapabilitiesDaoImpl.java [hunk 2]
```diff
@@ -106,4 +107,16 @@ public Boolean isVmSnapshotEnabled(HypervisorType hypervisorType, String hypervi
         HypervisorCapabilitiesVO result = getCapabilities(hypervisorType, hypervisorVersion);
         return result.getVmSnapshotEnabled();
     }
+
+    @Override
+    public List<HypervisorType> getHypervisorsWithDefaultEntries() {
```
Comment (anuragaw) [inline engine/schema/src/main/java/com/cloud/hypervisor/dao/HypervisorCapabilitiesDaoImpl.java]: Is there a possibility of high frequency calls to this method? If so we can perhaps optimize query calls with some caching? If not LGTM

Comment (nvazquez) [reply engine/schema/src/main/java/com/cloud/hypervisor/dao/HypervisorCapabilitiesDaoImpl.java]: Thanks @anuragaw, have refactored it to invoke this only once

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test matrix

**Comment by rhtyd** [comment]
Tests LGTM, they are failing irrespective of this PR.

**Pull request merged**